cgyle/cli.py

```diff
@@ -192,6 +193,32 @@ def update_cache(self) -> None:
                     if exception is not None:
                         logging.error(f'Thread failed with: {exception}')
 
+        # All done, collect errors if any. cgyle only keeps the
+        # log files of failed caching attempts and wipes the successful
+        # ones because there is no meaningful information in a successful
+        # caching process other than, the container was cached, which
+        # is an information that is still present by reading the log
+        # directory tree. The error information from all failed log
+        # files is now combined into one log file and appended to an
+        # eventually existing log file.
+        if not self.dryrun:
+            log_path = DistributionProxy.get_log_path()
+            log_file_name = f'{log_path}.log'
+            try:
+                with open(log_file_name, 'a+') as collect_fd:
+                    for topdir, dirs, files in sorted(os.walk(log_path)):
```

Comment:
We are walking `log_path` and it appears that `log_path` is also the place where the distribution container we use to fill the cache writes it's logs. Thus in the directory walking operation we will also pick up the log that is meant to be the collector, shouldn't we skip this file here? And maybe this is why the `a+` is used in the opening. But reading the collector log is not useful, we only want to read the log files that the distribution in the container leaves behind.

Reply:
The a+ mode was really just a mistake from my side and not related to the logic

log_path is exclusive to cgyle and currently set to ```/var/log/cgyle```. The collective log file name just appends ```.log``` to the logging directory. That is then ```/var/log/cgyle.log```. The collective log file is therefore not part of the os.walk

There is no other process that places data to the cgyle log directory and we collect after all threads that do the caching are completed.

> we only want to read the log files that the distribution in the container leaves behind

yes and this is what it does (also tested). 

Did I overlook something ?

Reply:
The user facing registry is configured via ```/etc/registry/config.yml``` and you can specify how it should do the logging in an extra ```log:``` section. At the moment we only set the logging level there which also means that our user facing registry only logs into the systemd journal and not into a file. If we change this in any case, the logging should **not** point to ```/var/log/cgyle``` because then your point is valid and we would collect strange things.

Overall the following is correct: If you place data to ```/var/log/cgyle``` while cgyle is in the collector processing it will pick up this data, if it belongs to cgyle or not. We could change this by passing the container paths from all threads as input to the collector but this increases the complexity of the implementation for questionable reasons

Reply:
I added a patch to the collector that only collects log files from containers that matches the catalog paths used for the caching. This also prevents that we accumulate the same error logs without a request

Reply:
> The a+ mode was really just a mistake from my side and not related to the logic
> 
> log_path is exclusive to cgyle and currently set to `/var/log/cgyle`. The collective log file name just appends `.log` to the logging directory. That is then `/var/log/cgyle.log`. The collective log file is therefore not part of the os.walk
> 
> There is no other process that places data to the cgyle log directory and we start collecting after all threads, that do the caching, are completed.
> 
> > we only want to read the log files that the distribution in the container leaves behind
> 
> yes and this is what it does (also tested).
> 
> Did I overlook something ?

No, I missed the differentiation between `var/log/cgyle` and `/var/log/cgyle.log`